smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/FrameworkErrorModel.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.smithy.typescript.codegen;
+
+import software.amazon.smithy.model.Model;
+import software.amazon.smithy.model.shapes.ShapeId;
+import software.amazon.smithy.model.shapes.StructureShape;
+import software.amazon.smithy.model.traits.ErrorTrait;
+import software.amazon.smithy.model.traits.HttpErrorTrait;
+
+public enum FrameworkErrorModel {
```

Comment:
At some point I want the user to be able to specify the namespace used for these framework errors (so that they can use the coral ones). I think it'd be best if they specified it at generation time; loading the model from a file and then mutating every resource in it seems more annoying than calling a bunch of builders. What benefit of a model file am I not thinking of?

Reply:
It's just really painfully tedious to create and edit a model in code vs in the IDL. Related, you should use a different namespace - `smithy.api` is a special namespace that's treated a little differently. `smithy.framework` perhaps?

Also related - why an enum instead of a static method? Some comments explaining these things would be nice.

Reply:
I will change the namespace. I'm still unconvinced that loading a model and then transforming all of the structures in it is less painful than just creating the model.

As for the enum, Effective Java 3rd ed., Item 3:
> a single-element enum type is often the best way to implement a singleton

(it guarantees single instantiation no matter what, which the typical singleton patterns don't)